### PR TITLE
Feat #109 - Allow making the audit parameter table optional

### PR DESCRIPTION
**Pull request by JiriBalcar**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -210,7 +210,7 @@ Non-blocking methods provide an `onError` consumer for custom error handling.
 
 The `audit-base` project provides auditing functionality for easier investigation of issues. Audit records are stored in a database and can be easily queried. The auditing library also handles removal of old audit records.
 
-The audit library requires two database tables. The DDL is available for the following databases:
+The audit library requires one database table `audit_log` and optionally second table `audit_params` for logging detail parameters. The DDL is available for the following databases:
```
[the remark this answers is not in the document]
Comment (JiriBalcar) [reply README.md]: fixed

File: audit-base/src/main/java/com/wultra/core/audit/base/database/DatabaseAuditWriter.java [hunk 2]
```diff
@@ -185,27 +189,32 @@ public int getBatchSize() {
...
-                                    ps.setTimestamp(2, new Timestamp(record.getTimestamp().getTime()));
-                                    ps.setString(3, StringUtil.trim(record.getKey(), 256));
-                                    Object value = record.getValue();
-                                    if (value == null) {
-                                        ps.setNull(4, Types.VARCHAR);
-                                    } else if (value instanceof CharSequence) {
-                                        ps.setString(4, StringUtil.trim(value.toString(), 4000));
-                                    } else {
-                                        ps.setString(4, StringUtil.trim(jsonUtil.serializeObject(value), 4000));
+                    if (paramLoggingEnabled) {
```
[the remark this answers is not in the document]
Comment (JiriBalcar) [reply audit-base/src/main/java/com/wultra/core/audit/base/database/DatabaseAuditWriter.java]: logic inverted

File: audit-base/src/test/java/com/wultra/core/audit/base/AuditParamEnabledTest.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
+package com.wultra.core.audit.base;
```
[the remark this answers is not in the document]
Comment (JiriBalcar) [reply audit-base/src/test/java/com/wultra/core/audit/base/AuditParamEnabledTest.java:16]: Added copyright header

File: audit-base/src/test/java/com/wultra/core/audit/base/AuditParamEnabledTest.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+import java.sql.Timestamp;
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@ExtendWith(SpringExtension.class)
```
[the remark this answers is not in the document]
Comment (JiriBalcar) [reply audit-base/src/test/java/com/wultra/core/audit/base/AuditParamEnabledTest.java]: Removed ExtendWith and removed public modifiers from test classes and test methods based on Sonar Lint reccomendation.

File: audit-base/src/main/java/com/wultra/core/audit/base/database/DatabaseAudit.java [hunk 5]
```diff
@@ -219,7 +219,7 @@ private void traceInternal(String message, AuditDetail detail, Object[] args) {
 
     @Override
     public boolean isLevelEnabled(@NonNull AuditLevel level) {
-        return configuration.getMinimumLevel().intValue() >= level.intValue();
+        return configuration.getMinimumLevel().intValue() <= level.intValue();
```
[the remark this answers is not in the document]
Comment (banterCZ) [reply audit-base/src/main/java/com/wultra/core/audit/base/database/DatabaseAudit.java:222]: No, thanks, you are not supposed to. We just agreed with @romanstrobl that we should improve it.

File: audit-base/src/test/java/com/wultra/core/audit/base/AuditTest.java [hunk 6]
```diff
@@ -62,6 +62,7 @@ public void testAuditInfo() {
         audit.info("test message");
         audit.flush();
         jdbcTemplate.query("SELECT * FROM audit_log", rs -> {
+            assertTrue(rs.next());
```
[the remark this answers is not in the document]
Comment (banterCZ) [reply audit-base/src/test/java/com/wultra/core/audit/base/AuditTest.java:57]: I do not want to push any my solution. I would try for example
```java
final SqlRowSet rs = jdbcTemplate.queryForRowSet("SELECT * FROM audit_log");
assertTrue(rs.next());
assertEquals("test-application", rs.getString("application_name"));
//...
```
It seems to me more straightforward than obfuscating via different lambdas. Anyway, this approach needs to handle assert of clob, I believe in you that you will find out the final solution.

Definitely it is not your fault, the original test code wasn't perfect.

[the remark this answers is not in the document]
Comment (romanstrobl) [reply audit-base/src/test/java/com/wultra/core/audit/base/AuditTest.java:57]: I would also prefer a solution without returning arbitrary values, it seems like a side effect. Thank you for fixing this old issue which was under the radar for us.

[the remark this answers is not in the document]
Comment (JiriBalcar) [reply audit-base/src/test/java/com/wultra/core/audit/base/AuditTest.java]: I used @banterCZ solution. Please check if the Clob handling is correct.

[the remark this answers is not in the document]
Comment (banterCZ) [reply audit-base/src/test/java/com/wultra/core/audit/base/AuditTest.java]: 👏

File: audit-base/src/test/java/com/wultra/core/audit/base/util/TestUtil.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
+package com.wultra.core.audit.base.util;
+
+import java.sql.Clob;
+
+public class TestUtil {
```
[the remark this answers is not in the document]
Comment (JiriBalcar) [reply audit-base/src/test/java/com/wultra/core/audit/base/util/TestUtil.java]: Fixed

File: audit-base/src/test/java/com/wultra/core/audit/base/util/TestUtil.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+     * Converts Clob to String.
+     *
+     * @param clobObject Clob.
+     * @return String.
+     */
+    public static String clobToString(Clob clobObject) {
+        try {
+            return clobObject.getSubString(1, Math.toIntExact(clobObject.length()));
+        } catch (Exception ex) {
+            throw new RuntimeException();
```
Comment (banterCZ) [inline audit-base/src/test/java/com/wultra/core/audit/base/util/TestUtil.java:38]: Pass the cause exception, please.

```suggestion
            throw new RuntimeException(ex);
```

**Review by banterCZ — APPROVED** [review]
Very good! Final merge is waiting for signing CLA.

**Comment by JiriBalcar** [comment]
> Very good! Final merge is waiting for signing CLA.

OK, where should I sign it?

**Comment by petrdvorak** [comment]
Hello @JiriBalcar, thank you - we mentioned the CLA just to keep everything clean. You can find the form here:
https://forms.gle/Au5aJDYf5J64jG4H6

**Comment by JiriBalcar** [comment]
> Hello @JiriBalcar, thank you - we mentioned the CLA just to keep everything clean. You can find the form here:
> https://forms.gle/Au5aJDYf5J64jG4H6

Form submitted

**Pull request merged**